packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -555,6 +557,29 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
             return;
         }
         this.term.open(this.node);
+
+        if (isFirefox) {
+            // monkey patching intersection observer handling for secondary window support
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            const renderService: any = (this.term as any)._core._renderService;
+            const originalFunc: (entry: IntersectionObserverEntry) => void = renderService._onIntersectionChange.bind(renderService);
+            const replacement = function (entry: IntersectionObserverEntry): void {
+                if (entry.target.ownerDocument !== document) {
+
+                    const patchedEvent: IntersectionObserverEntry = {
+                        ...entry,
+                        isIntersecting: true,
+                    };
+                    originalFunc(patchedEvent);
+                } else {
+                    originalFunc(entry);
+                }
+            };
+
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            renderService._onIntersectionChange = replacement;
```

Comment:
Please add a comment above the if-check to explain what the problem being solved is.

Alternative writing:

```suggestion
            const originalFunc: Function = renderService._onIntersectionChange;
            renderService._onIntersectionChange = function _onIntersectionChange(entry: IntersectionObserverEntry): void {
                // explain why this check is required and what it means...
                if (entry.target.ownerDocument !== document) {
                    const patchedEntry: IntersectionObserverEntry = {
                        ...entry,
                        isIntersecting: true,
                    };
                    originalFunc.call(this, patchedEntry);
                } else {
                    originalFunc.call(this, entry);
                }
            };
```

Reply:
As far as I can see, the difference is using `Function.call(thisArgument)` instead of binding the function. Any particular reason for that?

Reply:
I believe that the binding is unnecessary, because you assign the new function to a field on the original owner of the method, the `renderService`, so the `this` reference will remain the same. On the other hand, it's fine to `bind` to (1) show your intention to keep the same `this` and (2) be safe in case anyone later uses `originalFunc` in a different way and doesn't check the binding - a risk that Paul's suggestion makes real :-).

Reply:
It made more sense to me to write the replacement function as a regular method where `this` is bound to its owner, the `renderService` object here. But ultimately both writing achieve the same result so it's fine.

You added the comment as I requested and I'm plenty happy with it now :)